### PR TITLE
Parse default-host-header property properly

### PR DESCRIPTION
**Pull request by kubukoz**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

## Purpose

This PR changes the handling of `default-host-header` in akka-http's config that Play uses to run a server in `play-akka-http-server`. Previously, the value of the property was passed as the hostname to the `Host` class's companion's `apply` method, which could be invalid if the value contained a hostname with a port. Now it'll delegate parsing to akka-http's header parsing and handle ports properly.

## References

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

### CODE REVIEW

File: transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.core.server.akkahttp
+
+import play.api.Configuration
+import akka.http.scaladsl.model.headers.Host
+
+class AkkaServerConfigReader(akkaServerConfig: Configuration) {
```
Comment (kubukoz) [inline transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala]: I introduced this class to make it possible to test the new behavior. If you think it's not worth it as we're delegating to akka-http, I'll bring the logic back to the other class (although we were delegating previously and the bug was still there) :)

File: transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import play.api.Configuration
+import akka.http.scaladsl.model.headers.Host
+
+private[play] final class AkkaServerConfigReader(akkaServerConfig: Configuration) {
+  def getHostHeader: Either[Throwable, Host] = {
+    Host
+      .parseFromValueString(akkaServerConfig.get[String]("default-host-header"))
+      .left
+      .map { errors =>
+        new RuntimeException(
```
Comment (marcospereira) [inline transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala]: Use `Configuration.reportError` which gives better error reporting.

File: transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.core.server.akkahttp
+
+import play.api.Configuration
+import akka.http.scaladsl.model.headers.Host
+
+private[play] final class AkkaServerConfigReader(akkaServerConfig: Configuration) {
```
Comment (marcospereira) [inline transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala]: Make it `private[server]`.

File: transport/server/play-akka-http-server/src/test/scala/play/core/server/akkahttp/AkkaServerConfigReaderTest.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    "parse Host header with port number" in {
+      val reader = new AkkaServerConfigReader(Configuration("default-host-header" -> "localhost:4000"))
+      val actual = reader.getHostHeader
+
+      actual mustEqual Right(Host("localhost", 4000))
+    }
+
+    "parse Host header with port number" in {
```
Comment (marcospereira) [inline transport/server/play-akka-http-server/src/test/scala/play/core/server/akkahttp/AkkaServerConfigReaderTest.scala]: Have another description for this test.

**Comment by mkurz** [comment]
@kubukoz The build fails: https://travis-ci.com/playframework/playframework/jobs/190348450#L2005-L2010
```
[error] /home/travis/build/playframework/playframework/transport/server/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaServerConfigReader.scala:19:11: type mismatch;
[error]  found   : RuntimeException
[error]  required: Option[Throwable]
[error]           new RuntimeException(errors.map(_.formatPretty).mkString(", "))
[error]           ^
[error] one error found
```

**Comment by mkurz** [comment]
@kubukoz If you can fix this we may be able to merge this into upcoming 2.7.1.

**Comment by kubukoz** [comment]
I'll fix it today! Sorry, I was busy conferencing :)

Any chance of a backport released to 2.6 if I contribute one?

**Comment by mkurz** [comment]
> Any chance of a backport released to 2.6 if I contribute one?

**Comment by kubukoz** [comment]
Okay, finally got a moment to actually fix the compilation... hopefully that's it.

Turns out I don't need this change in 2.6 anymore, so to make the situation less messy I suggest to just keep it for 2.7.2 :)

**Pull request merged**